researchspace/researchspace
# Re-enable linked viewes in the Thinking Frames.

# Why

Linked views in Thinking Frames were not working properly.

# What

Changes to `DashboardComponent` to re-enable linked views. Linked views are implemented as a nested FlexLayout element:  
- individual tabs in linked views can't be closed or moved around
- one can't create a new empty tab inside nested view
- to close nested tabs one need to close parent tab

Added new view type "hidden". Should be used together with linked views when view config shouldn't be displayed in the UI on its own, but is used only through linked views.  

Also when tab was closed the actual state of the component was not updated accordingly. So when one close/open tabs in the UI corresponding state array was never cleaned up. This change properly removes item from `state.items` when tabs are closed. 

# Meta
I placed nested tab at the bottom of the screen, to somehow differentiate them from the top level tabs. I guess one can also change nested tab color to make it more distinct.

# Video / Gif / Screenshot

![localhost_10214_resource_ThinkingFramesTest](https://github.com/researchspace/researchspace/assets/2255044/0d1bdb76-b46e-40ba-ac0f-52fd8da5c20f)

# How To Test
There are some linked view configs in the default templates, they should work as before. Like "Image graph authoring".

Or you can just link any of the pre-defined views, e.g:
```
    linked-views='[{
      "id": "test",
      "label": "Linked Test",
      "iconName": "wallpaper",
      "viewIds": ["resource", "iiif"],
      "resourceNotRequired": false,
      "type": "view"
    }
    ]'
```
